lib/segment/src/common/utils.rs

```diff
@@ -316,6 +317,109 @@ pub fn remove_value_from_json_map(
     }
 }
 
+/// Insert value at a given JSON path  in JSON array
+pub fn insert_at_array_path(
+    array_path: &str,
+    array_index: Option<u32>,
+    rest_path: Option<&str>,
+    value: &mut serde_json::Map<String, Value>,
+    new_value: Value,
+) {
+    match value.get_mut(array_path) {
```

Comment:
Thanks I did not know about the performance aspect of the entry API :+1:  

The issue with it is that I have written the entire JsonPath internals with `get/get_mut` so I'd rather keep it the same way for consistency in this PR.

However I think it would make sense to migrate the whole file at once to the entry API in different PR.
WDYT?

Reply:
Rust is full of magical stuff. The single-lookup mechanism is basically the reason the entry API exists, it's not just for functional style usage. :smiley: 

> However I think it would make sense to migrate the whole file at once to the entry API in a different PR.
WDYT?

I'd prefer that for the above reason, yes.

Maybe we can first conclude how hot this code actually is before deciding on whether to make this change. It's definitely not critical.

Reply:
> Maybe we can first conclude how hot this code actually is before deciding on whether to make this change. It's definitely not critical.

This code is being used more and more but I'd prefer not changing it until we have a real performance issue about it.

Now we just need to find a good way to capture the knowledge, either a comment in the code or a Github issue.
I would favor the comment as there is not action required right now.

Reply:
I propose https://github.com/qdrant/qdrant/commit/6c674ff1587d888670009a87bef8fe2d87fce56a

I picked the two important functions to cover `get` & `get_mut`.